main.tf

```diff
@@ -763,6 +765,66 @@ data "aws_iam_policy_document" "access_log_delivery" {
   }
 }
 
+# Grant read and list access to CloudFront Origin Access Identities (deprecated by AWS)
+data "aws_iam_policy_document" "cloudfront_oai_read_policy" {
+  count = local.create_bucket && var.attach_cloudfront_oai_read_policy ? 1 : 0
+
+  statement {
+    sid = "AWSCloudFrontOAIRead"
```

Comment:
Please use the same `sid` as the one you are copying it from the documentation (e.g. `PolicyForCloudFrontPrivateContent`).

Reply:
Let's change this statement to support `write` actions as well. Let's use boolean control named `var.attach_cloudfront_oai_policy` and allow parameterization of IAM actions used by OAI via a separate variable:
```
cloudfront_oai_iam_actions = ["s3:GetObject", "s3:PutObject"]
```

The default value is of `cloudfront_oai_iam_actions` is `["s3:GetObject"]`